src/main/java/io/github/intellij/dlanguage/psi/DlangFile.java

```diff
@@ -48,10 +51,19 @@ public Icon getIcon(final int flags) {
     @Nullable
     public String getModuleName() {
         final DLanguageModuleDeclaration module = findChildByClass(DLanguageModuleDeclaration.class);
+
         if (module == null) {
             return null;
         }
-        return module.getText().replaceAll(";", "").replaceAll("^module\\s+", "");
+
+        String[] segments = module.getText()
+                            .substring(6, module.getTextLength() - 1)
+                            .split("\\.");
+
+        return Optional.of(segments)
+                       .filter(ArrayUtils::isEmpty)
+                       .map(sc -> sc[segments.length - 1])
+                       .orElse(null);
```

Comment:
Why not just keep this as `return module.getText().replaceAll(";", "").replaceAll("^module\\s+", "");`?

Reply:
I think comment mentioned obsolete change. The main reason is that it is more npe safe and instead of rebuilding module declaration and then stripping unnecessary parts, this version just searches for identifier in bultin ast tree, and uses it. Imho it is more direct approch rather than original one.

I'll double check this method as it seems it is doing not it is intended to.

Just for info, new to intellij platform so if something is wrong about new code interacting with intellij idea please mention it.